### PR TITLE
various: cleanup of "inherit version;"

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
a bit related to #103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
```
error: undefined variable 'version'                                                       
                                                                                          
       at /home/siraben/.cache/nixpkgs-review/pr-130472/nixpkgs/pkgs/development/libraries/clfft/default.nix:10:25:
                                             
            9|     repo = "clFFT";                                                                                                                                                   
           10|     rev = "refs/tags/v${version}";                                                                                                                                    
             |                         ^                                                  
           11|     sha256 = "134vb6214hn00qy84m4djg4hqs6hw19gkp8d0wlq8gb9m3mfx7na";
                                                                                          
       … while evaluating 'makeOverridable'     
```

**Comment by Stunkymonkey** [comment]
@siraben please evaluate again now.

**Comment by siraben** [comment]
Ok, it's going to take a while (several hours). Evaluation seems to be working.

File: pkgs/development/libraries/nss/3.53.nix [hunk 1]
```diff
@@ -4,23 +4,20 @@
...
-  inherit version;
+  version = "3.53.1";
 
   src = fetchurl {
-    url = "mirror://mozilla/security/nss/releases/NSS_${underscoreVersion}_RTM/src/${pname}-${version}.tar.gz";
+    url = "mirror://mozilla/security/nss/releases/NSS_${lib.replaceStrings ["."] ["_"] version}_RTM/src/${pname}-${version}.tar.gz";
     sha256 = "05jk65x3zy6q8lx2djj8ik7kg741n88iy4n3bblw89cv0xkxxk1d";
   };
 
+  nssPEM = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nss/3.53.nix:16]: I think if you move this back into the let in we can avoid some rebuilds.

Comment (Stunkymonkey) [reply pkgs/development/libraries/nss/3.53.nix:16]: yes this a good idea

File: pkgs/development/libraries/java/rhino/default.nix [hunk 2]
```diff
@@ -1,23 +1,19 @@
...
 
   src = fetchurl {
     url = "mirror://mozilla/js/rhino1_7R2.zip";
     sha256 = "1p32hkghi6bkc3cf2dcqyaw5cjj7403mykcp0fy8f5bsnv0pszv7";
   };
 
+  xbeans  = fetchurl {
+    url = "http://archive.apache.org/dist/xmlbeans/binaries/xmlbeans-2.2.0.zip";
+    sha256 = "1pb08d9j81d0wz5wj31idz198iwhqb7mch872n08jh1354rjlqwk";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/rhino/default.nix:15]: Can you move this back, too?

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130472` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages marked as broken and skipped:</summary>
  <ul>
    <li>authenticator</li>
    <li>gnome.gnome-documents</li>
    <li>libsForQt512.dragon</li>
    <li>libsForQt512.kamoso</li>
    <li>libsForQt512.kdenlive</li>
    <li>libsForQt512.phonon-backend-gstreamer</li>
    <li>libsForQt514.dragon</li>
    <li>libsForQt514.kamoso</li>
    <li>libsForQt514.kdenlive</li>
    <li>odpdown</li>
    <li>openmodelica</li>
    <li>termplay</li>
    <li>xmonad_log_applet</li>
  </ul>
</details>
<details>
  <summary>21 packages failed to build:</summary>
  <ul>
    <li>citrix_workspace (citrix_workspace_21_06_0)</li>
    <li>citrix_workspace_20_04_0</li>
    <li>citrix_workspace_20_06_0</li>
    <li>citrix_workspace_20_09_0</li>
    <li>citrix_workspace_20_10_0</li>
    <li>citrix_workspace_20_12_0</li>
    <li>citrix_workspace_21_01_0</li>
    <li>citrix_workspace_21_03_0</li>
    <li>gnome.gnome-todo</li>
    <li>home-assistant</li>
    <li>hunter</li>
    <li>ib-controller</li>
    <li>kooha</li>
    <li>lexicon</li>
    <li>libadwaita</li>
    <li>mousai</li>
    <li>notejot</li>
    <li>radiotray-ng</li>
    <li>shortwave</li>
    <li>xiphos</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>
<details>
  <summary>349 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>alfis</li>
    <li>almanah</li>
    <li>apacheHttpdPackages.mod_auth_mellon (apacheHttpdPackages_2_4.mod_auth_mellon)</li>
    <li>apostrophe</li>
    <li>aravis</li>
    <li>areca</li>
    <li>astroid</li>
    <li>audio-recorder</li>
    <li>aws-workspaces</li>
    <li>azureus</li>
    <li>balsa</li>
    <li>baresip</li>
    <li>birdfont</li>
    <li>bookworm</li>
    <li>brasero</li>
    <li>brasero-original</li>
    <li>bubblemail</li>
    <li>byzanz</li>
    <li>calls</li>
    <li>cawbird</li>
    <li>chrome-gnome-shell</li>
    <li>cinnamon.cinnamon-common</li>
    <li>cinnamon.cinnamon-control-center</li>
    <li>cinnamon.cinnamon-gsettings-overrides</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>claws-mail</li>
    <li>cog</li>
    <li>denemo</li>
    <li>dropbox-cli</li>
    <li>easyeffects</li>
    <li>eclipse-mat</li>
    <li>eclipses.eclipse-committers</li>
    <li>eclipses.eclipse-cpp</li>
    <li>eclipses.eclipse-java</li>
    <li>eclipses.eclipse-jee</li>
    <li>eclipses.eclipse-modeling</li>
    <li>eclipses.eclipse-platform</li>
    <li>eclipses.eclipse-rcp</li>
    <li>eclipses.eclipse-scala-sdk</li>
    <li>eclipses.eclipse-sdk</li>
    <li>elementary-planner</li>
    <li>empathy (gnome.empathy)</li>
    <li>enlightenment.rage</li>
    <li>eolie</li>
    <li>ephemeral</li>
    <li>epiphany (gnome.epiphany)</li>
    <li>evince (gnome.evince)</li>
    <li>evolution</li>
    <li>evolution-data-server (gnome.evolution-data-server ,gnome.evolution_data_server)</li>
    <li>evolution-ews</li>
    <li>evolutionWithPlugins</li>
    <li>farstream</li>
    <li>fcitx5-gtk</li>
    <li>fcitx5-with-addons</li>
    <li>feedreader</li>
    <li>firefox-esr-78-unwrapped (firefoxPackages.firefox-esr-78)</li>
    <li>foliate</li>
    <li>folks (gnome.folks)</li>
    <li>font-manager</li>
    <li>formiko</li>
    <li>fractal</li>
    <li>gajim</li>
    <li>gcompris</li>
    <li>gfbgraph (gnome.gfbgraph)</li>
    <li>giara</li>
    <li>glade (gnome.glade)</li>
    <li>glom</li>
    <li>gmrender-resurrect</li>
    <li>gnome-builder</li>
    <li>gnome-feeds</li>
    <li>gnome-inform7</li>
    <li>gnome-keysign</li>
    <li>gnome-network-displays</li>
    <li>gnome-online-accounts (gnome.gnome-online-accounts ,gnome.gnome_online_accounts)</li>
    <li>gnome-photos (gnome.gnome-photos)</li>
    <li>gnome-podcasts</li>
    <li>gnome-recipes</li>
    <li>gnome.bijiben (gnome.gnome-notes)</li>
    <li>gnome.cheese</li>
    <li>gnome.devhelp</li>
    <li>gnome.file-roller</li>
    <li>gnome.geary</li>
    <li>gnome.gnome-applets</li>
    <li>gnome.gnome-books</li>
    <li>gnome.gnome-boxes</li>
    <li>gnome.gnome-calendar</li>
    <li>gnome.gnome-chess</li>
    <li>gnome.gnome-contacts</li>
    <li>gnome.gnome-control-center (gnome.gnome_control_center)</li>
    <li>gnome.gnome-flashback</li>
    <li>gnome.gnome-initial-setup</li>
    <li>gnome.gnome-maps</li>
    <li>gnome.gnome-music</li>
    <li>gnome.gnome-online-miners</li>
    <li>gnome.gnome-panel</li>
    <li>gnome.gnome-session (gnome.gnome_session)</li>
    <li>gnome.gnome-shell (gnome.gnome_shell)</li>
    <li>gnome.gnome-software</li>
    <li>gnome.gnome-sound-recorder</li>
    <li>gnome.gnome-terminal (gnome.gnome_terminal)</li>
    <li>gnome.gnome-tweaks (gnome.gnome-tweak-tool)</li>
    <li>gnome.gnome-user-share</li>
    <li>grilo-plugins (gnome.grilo-plugins)</li>
    <li>gnome.gvfs</li>
    <li>libgdata (gnome.libgdata)</li>
    <li>libgepub (gnome.libgepub)</li>
    <li>libzapojit (gnome.libzapojit)</li>
    <li>gnome.nautilus</li>
    <li>gnome.nautilus-python</li>
    <li>gnome.polari</li>
    <li>gnome.pomodoro</li>
    <li>gnome.rygel</li>
    <li>shotwell (gnome.shotwell)</li>
    <li>gnome.sushi</li>
    <li>gnome.totem</li>
    <li>tracker-miners (gnome.tracker-miners)</li>
    <li>webkitgtk (gnome.webkitgtk)</li>
    <li>yelp (gnome.yelp)</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.night-theme-switcher</li>
    <li>gnucash</li>
    <li>gnunet-gtk</li>
    <li>gnvim</li>
    <li>gnvim-unwrapped</li>
    <li>gscrabble</li>
    <li>gst_all_1.gst-editing-services</li>
    <li>gst_all_1.gst-plugins-bad</li>
    <li>gst_all_1.gst-rtsp-server</li>
    <li>gst_all_1.gst-vaapi</li>
    <li>gthumb</li>
    <li>gtk4</li>
    <li>gtkd</li>
    <li>gtkmm4</li>
    <li>gtksourceview5</li>
    <li>haskellPackages.ghcjs-dom-hello</li>
    <li>haskellPackages.gi-javascriptcore</li>
    <li>haskellPackages.gi-webkit2</li>
    <li>haskellPackages.gi-webkit2webextension</li>
    <li>haskellPackages.jsaddle-webkit2gtk</li>
    <li>haskellPackages.reflex-dom</li>
    <li>haskellPackages.reflex-dom-fragment-shader-canvas</li>
    <li>haskellPackages.reflex-localize-dom</li>
    <li>haskellPackages.webkit2gtk3-javascriptcore</li>
    <li>helvum</li>
    <li>hqplayerd</li>
    <li>ipscan</li>
    <li>kaidan</li>
    <li>kgx</li>
    <li>kicad-small</li>
    <li>kicad-unstable-small</li>
    <li>kmymoney</li>
    <li>kotatogram-desktop</li>
    <li>lasso</li>
    <li>libofa</li>
    <li>libreoffice (libreoffice-still)</li>
    <li>libreoffice-fresh</li>
    <li>libreoffice-fresh-unwrapped</li>
    <li>libreoffice-qt</li>
    <li>libreoffice-unwrapped (libreoffice-still-unwrapped)</li>
    <li>libsForQt5.dragon (libsForQt515.dragon ,plasma5Packages.dragon)</li>
    <li>libsForQt5.kamoso (libsForQt515.kamoso ,plasma5Packages.kamoso)</li>
    <li>libsForQt5.kdenlive (libsForQt515.kdenlive ,plasma5Packages.kdenlive)</li>
    <li>libsForQt5.phonon-backend-gstreamer (libsForQt515.phonon-backend-gstreamer ,plasma5Packages.phonon-backend-gstreamer)</li>
    <li>libsForQt5.telepathy (libsForQt515.telepathy ,plasma5Packages.telepathy)</li>
    <li>libsForQt512.telepathy</li>
    <li>libsForQt514.phonon-backend-gstreamer</li>
    <li>libsForQt514.telepathy</li>
    <li>liferea</li>
    <li>lispPackages.cl-webkit2</li>
    <li>lispPackages.nyxt</li>
    <li>lollypop</li>
    <li>luakit</li>
    <li>lutris</li>
    <li>lutris-free</li>
    <li>lutris-unwrapped</li>
    <li>mailnag</li>
    <li>mailnagWithPlugins</li>
    <li>mapmap</li>
    <li>marker</li>
    <li>mate.atril</li>
    <li>mate.mate-user-guide</li>
    <li>matrix-synapse</li>
    <li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
    <li>mavproxy</li>
    <li>megapixels</li>
    <li>meteo</li>
    <li>midori-unwrapped</li>
    <li>mmex</li>
    <li>mopidy-iris</li>
    <li>mopidy-local</li>
    <li>mopidy-mopify</li>
    <li>mopidy-mpd</li>
    <li>mopidy-mpris</li>
    <li>mopidy-musicbox-webclient</li>
    <li>mopidy-podcast</li>
    <li>mopidy-scrobbler</li>
    <li>mopidy-somafm</li>
    <li>mopidy-soundcloud</li>
    <li>mopidy-spotify-tunigo</li>
    <li>mopidy-subidy</li>
    <li>mopidy-tunein</li>
    <li>mopidy-youtube</li>
    <li>nasc</li>
    <li>newsflash</li>
    <li>nheko</li>
    <li>notes-up</li>
    <li>nss_3_53</li>
    <li>nyxt</li>
    <li>onboard</li>
    <li>osmo</li>
    <li>pantheon.elementary-calendar</li>
    <li>pantheon.elementary-camera</li>
    <li>pantheon.elementary-capnet-assist</li>
    <li>pantheon.elementary-code</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-gsettings-schemas</li>
    <li>pantheon.elementary-mail</li>
    <li>pantheon.elementary-music</li>
    <li>pantheon.elementary-photos</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-videos</li>
    <li>pantheon.extra-elementary-contracts</li>
    <li>pantheon.notes-up</li>
    <li>pantheon.switchboard-plug-a11y</li>
    <li>pantheon.switchboard-plug-onlineaccounts</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>parlatype</li>
    <li>pdfpc</li>
    <li>phosh</li>
    <li>picard</li>
    <li>pidgin</li>
    <li>pidgin-carbons</li>
    <li>pidgin-latex</li>
    <li>pidgin-mra</li>
    <li>pidgin-msn-pecan</li>
    <li>pidgin-opensteamworks</li>
    <li>pidgin-osd</li>
    <li>pidgin-otr</li>
    <li>pidgin-sipe</li>
    <li>pidgin-skypeweb</li>
    <li>pidgin-window-merge</li>
    <li>pidgin-xmpp-receipts</li>
    <li>pipelight</li>
    <li>pithos</li>
    <li>pitivi</li>
    <li>playonlinux</li>
    <li>portfolio</li>
    <li>powerdns-admin</li>
    <li>pragha</li>
    <li>printrun</li>
    <li>pulseeffects-legacy</li>
    <li>purple-discord</li>
    <li>purple-facebook</li>
    <li>purple-hangouts</li>
    <li>purple-lurch</li>
    <li>purple-matrix</li>
    <li>purple-mm-sms</li>
    <li>purple-plugin-pack</li>
    <li>purple-slack</li>
    <li>purple-vk-plugin</li>
    <li>purple-xmpp-http-upload</li>
    <li>python38Packages.humblewx</li>
    <li>python38Packages.kicad</li>
    <li>python38Packages.kivy</li>
    <li>python38Packages.pysaml2</li>
    <li>python38Packages.python3-saml</li>
    <li>python38Packages.sunpy</li>
    <li>python38Packages.total-connect-client</li>
    <li>python38Packages.wxPython_4_0</li>
    <li>python38Packages.wxPython_4_1</li>
    <li>python38Packages.xmlsec</li>
    <li>python38Packages.zeep</li>
    <li>python39Packages.humblewx</li>
    <li>python39Packages.kicad</li>
    <li>python39Packages.kivy</li>
    <li>python39Packages.pysaml2</li>
    <li>python39Packages.python3-saml</li>
    <li>python39Packages.sunpy</li>
    <li>python39Packages.total-connect-client</li>
    <li>python39Packages.wxPython_4_0</li>
    <li>python39Packages.wxPython_4_1</li>
    <li>python39Packages.xmlsec</li>
    <li>python39Packages.zeep</li>
    <li>pytrainer</li>
    <li>qgroundcontrol</li>
    <li>qt-video-wlr</li>
    <li>quilter</li>
    <li>quisk</li>
    <li>quodlibet</li>
    <li>quodlibet-full</li>
    <li>quodlibet-without-gst-plugins</li>
    <li>quodlibet-xine</li>
    <li>quodlibet-xine-full</li>
    <li>qutebrowser</li>
    <li>qxmpp</li>
    <li>recapp</li>
    <li>rednotebook</li>
    <li>remmina</li>
    <li>rhino</li>
    <li>run-scaled</li>
    <li>rymcast</li>
    <li>sayonara</li>
    <li>setzer</li>
    <li>skype4pidgin</li>
    <li>skytemple</li>
    <li>sound-juicer</li>
    <li>sparkleshare</li>
    <li>subtitleeditor</li>
    <li>surf</li>
    <li>surf-display</li>
    <li>swt</li>
    <li>swt_jdk8</li>
    <li>tangram</li>
    <li>tdesktop</li>
    <li>telegram-purple</li>
    <li>telepathy-farstream</li>
    <li>telepathy-haze</li>
    <li>thunderbird (thunderbird-78)</li>
    <li>tilix</li>
    <li>timeline</li>
    <li>tonelib-zoom</li>
    <li>toxprpl</li>
    <li>transcribe</li>
    <li>tunefish</li>
    <li>tuxguitar</li>
    <li>ulauncher</li>
    <li>unoconv</li>
    <li>vala-language-server</li>
    <li>vimb-unwrapped</li>
    <li>vocal</li>
    <li>vokoscreen-ng</li>
    <li>vuze</li>
    <li>webkit2-sharp</li>
    <li>whatsapp-for-linux</li>
    <li>wine (winePackages.full)</li>
    <li>wine-staging</li>
    <li>wineWowPackages.full</li>
    <li>winetricks</li>
    <li>wrapGAppsHook4</li>
    <li>xfce.parole</li>
    <li>xmlsec</li>
    <li>xplayer</li>
    <li>xpra</li>
    <li>yaxg</li>
  </ul>
</details>

**Pull request merged**